### PR TITLE
Aa/update on grafana and firewall

### LINKED ISSUE #5: Configure firewall rules
## Goal

Check and configure all necessary firewall rules.

Refer to https://containerssh.io/guides/honeypot/

### LINKED ISSUE #38: Update Monitoring on Grafana
Grafana:
-> show available and usage disk - at the same time
-> set up home dashboard
-> fix no data when a VM is down

### PR DESCRIPTION
**Pull request by bafread**
1. Secure Prometheus, so it's not reachable from outside
2. Small Change on Firewall rules (fix wrong port number on cadvisor-sacrificial-vm)
3. Update Monitoring: add requested data to monitor, add Home-Dashboard and fix issue with no data if server offline 

This PR will
- close #38 #39  
- fix no data when a server offline

- Terraform apply and login on the grafana at `logger-vm:3000`
  
## Server Online
Screenshot from 2022-07-03 23-55-45

## 5 Minute after Server Offline
Screenshot from 2022-07-04 00-04-10

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -83,14 +78,14 @@ resource "google_compute_firewall" "firewall_sacrificial_exception" {
...
-# open sacrificial-port 9100 to our prometheus
+# open sacrificial-port 8088 for cadvisor and 9100 for node-exporter
 resource "google_compute_firewall" "firewall_sacrificial_nodeexport" {
   name    = "firewall-sacrificial-nodeexport"
   network = google_compute_network.main.self_link
 
   allow {
     protocol = "tcp"
-    ports    = ["8080", "9100"]
+    ports    = ["8088", "9100"]
```
Comment (bafread) [inline terraform/main.tf:88]: Fix wrong port (cadvisor in Sacrificial-VM)

File: terraform/scripts/run_grafana.sh [hunk 2]
```diff
@@ -13,6 +13,7 @@ sudo docker run \
   --volume=grafana_data:/var/lib/grafana \
   --volume=/home/deployer/grafana/provisioning/datasources:/etc/grafana/provisioning/datasources \
   --volume=/home/deployer/grafana/provisioning/dashboards:/etc/grafana/provisioning/dashboards \
+  --env "GF_DASHBOARDS_DEFAULT_HOME_DASHBOARD_PATH=/etc/grafana/provisioning/dashboards/containerSSH.json" \
```
Comment (bafread) [inline terraform/scripts/run_grafana.sh:16]: it's for grafana default home dashboard

File: terraform/main.tf [hunk 3]
```diff
@@ -23,34 +23,29 @@ resource "google_compute_subnetwork" "honeypot_subnet" {
...
 
-  allow {
-    protocol = "udp"
-    ports    = ["0-65535"]
-  }
-
   allow {
     protocol = "tcp"
-    ports    = ["0-65535"]
+    ports    = ["22"]
```
Comment (paseaf) [inline terraform/main.tf:36]: We also need to allow port `2333` for SSH server on gateway VM, so we can SSH into it for development,  and in the end disable port `22` from external network on sacrificial VM.

Maybe keep this `allow-all` for now. We can remove it as part of #5 so we can verify firewall rules at once

**Review by paseaf — CHANGES_REQUESTED** [review]
LGTM! only one small change

File: terraform/main.tf (hunk 3 above)
Comment (bafread) [reply terraform/main.tf:36]: Ah ok, then i'll recover the `allow-all` rule. I'm just afraid that if we open all ports then we will also get some attacks or something similar, that are not relevant for our Honeypot. ^^

**Pull request merged**